src/Microsoft.AspNet.Http.Abstractions/Extensions/UseMiddlewareExtensions.cs

```diff
@@ -24,15 +24,28 @@ public static IApplicationBuilder UseMiddleware(this IApplicationBuilder builder
             {
                 var instance = ActivatorUtilities.CreateInstance(builder.ApplicationServices, middleware, new[] { next }.Concat(args).ToArray());
                 var methodinfo = middleware.GetMethod("Invoke", BindingFlags.Instance | BindingFlags.Public);
+                if (methodinfo == null)
+                {
+                    throw new InvalidOperationException("No public Invoke method found");
```

Comment:
End sentences with periods. Also, we might want to start using a resx file here (the `Microsoft.AspNet.Http.Extensions` project already uses one).